doc/SCALING.md

```diff
@@ -0,0 +1,106 @@
+# Scaling
+
+## Table of Contents
+
+- [Why](#why)
+- [Scaling with STUNner](#scaling-with-stunner)
+  - [Scaling-up (out)](#scaling-up-out)
+  - [Scaling-down (in)](#scaling-down-in)
+  - [Graceful shutdown](#more-on-the-graceful-shutdown)
+- [Example](#example)
+
+## Why
+
+Autoscaling is one of the key features in Kubernetes cluster. It is a feature in which the cluster is capable of increasing the number of pods as the demand for service increases and decrease the number of pods as the requirement decreases.
+
+The biggest advantage of using Kubernetes for autoscaling is, it reduces the cost and saves you from a lot of head scratching. You do not need to guess the number of nodes or pods needed to run your workloads. It scales up or down dynamically based on resource utilization, thus saving you dollars.
+
+When defining a Kubernetes deployment's resource limits and requests, it is a better practice to *not* go for an indefinite amount of CPUs but to keep the limits down and scale out (increase) the number of running pods if needed. Scaling a service might be scary and complex, this guide aims to overcome the fear.
+
+And there are other factors that might boost your need for scaling:
+- smaller load on each instance, might result in better, more stable performance
+- for some reason an application instance dies, the less call will be affected in the ICE reconnect
+
+
+## Scaling with STUNner
+
+STUNner has a full support for horizontal scaling. That means the number of pods can be increased and decreased according to the user's needs. 
+In case the user wants to scale the instances of the `stunner` deployment, the Kubernetes built in `HorizontalPodAutoscaler` can be used. The triggering event can be based on different metrics but it is advised to use the currently utilized amount of CPU. If so, it is important to state that the HPA [uses the requested amount of resources](https://pauldally.medium.com/horizontalpodautoscaler-uses-request-not-limit-to-determine-when-to-scale-97643d808997) to determine when to scale-up or down the number of instances.
+
+###  Scaling-up (out)
+
+When adding new instances to the existing `stunner` replica set we don't have much to worry about. The infrastructure for existing calls won't get interrupted, it stays the same as before the upscale event. Only when new user calls come in the cloud loadbalancer have an additional endpoint to choose from. This way we can achieve that STUNner is never going to be the bottleneck in the system. Obviously, if you have the needed computational power under your cluster.
```

Comment:
Not sure how to change this. In this section i was focusing on scaling up. Tried to separate the two, because they follow an extremely different approach.

Reply:
BTW when scaling up, the existing stunner infra does not change, only a new instance will come up. Wdym by my last test?